### ISSUE TITLE
pytorch 1.1.0 fails to load on windows (python3.6, 3.7)

### ISSUE THREAD

**Issue by jvesely** [pytorch/pytorch#21363]
## 🐛 Bug

`import torch` fails on windows using pytorch-1.1.0

## To Reproduce

1. import torch
```
_______ ERROR collecting tests/composition/test_autodiffcomposition.py ________
tests\composition\test_autodiffcomposition.py:18: in <module>
    import torch
C:\Users\appveyor\AppData\Roaming\Python\Python37\site-packages\torch\__init__.py:79: in <module>
    from torch._C import *
E   RuntimeError: generic_type: cannot initialize type "TensorProtoDataType": an object with that name is already defined
```

## Expected behavior

successfull import

## Environment
appveyor python3.6 and 3.7
pytorch installed via:
`pip install --user http://download.pytorch.org/whl/cpu/torch-1.1.0-cp37-cp37m-win_amd64.whl`

**Comment by jvesely** [comment]
pytorch 1.0.1 imports OK

**Comment by peterjc123** [comment]
@ezyang Do you know where the type `TensorProtoDataType` is initialized?

**Comment by ezyang** [comment]
I believe it's referring to `torch/csrc/onnx/init.cpp`

@jvesely, does this happen to you if you have a completely fresh Python environment, and install PyTorch from there? My easiest guess is that you somehow have two copies of PyTorch floating around here.

**Comment by jvesely** [comment]
> I believe it's referring to `torch/csrc/onnx/init.cpp`
> 
> @jvesely, does this happen to you if you have a completely fresh Python environment, and install PyTorch from there?
yes. The problem appears in appveyor CI [0, 1.0.1: 1], everything is installed from scratch.

[0] https://ci.appveyor.com/project/jvesely/psyneulink-wuxsn/builds/25037488/job/28d6ddsrh2hga5sb?fullLog=true
[1] https://ci.appveyor.com/project/jvesely/psyneulink-wuxsn/builds/25042790/job/bevusdmjgmn289bx?fullLog=true

**Comment by peterjc123** [comment]
I saw that you are using `pip install --user`. Is it possible that there is one installed in the system?

**Comment by jvesely** [comment]
> I saw that you are using `pip install --user`. Is it possible that there is one installed in the system?

It's possible, but I don't think it's the case.
Removing the explicit install line results in pip downloading the sources and failing to compile the torch package [0] (it looks like a missing build time dependency).

Not installing pytorch at all makes all the test pass (pytorch is an optional dep), but the number of skipped tests increases from 582 to 606[1]

[0] https://ci.appveyor.com/project/jvesely/psyneulink-wuxsn/builds/25469388/job/mnxthwl4al7s68in#L300
[1] https://ci.appveyor.com/project/jvesely/psyneulink-wuxsn/builds/25469472/job/mi2jky0aey53rh5o#L5362

**Comment by peterjc123** [comment]
How did you try to install? pip install torch==1.1? That is not correct. You should refer to pytorch.org.

**Comment by jvesely** [comment]
> How did you try to install? pip install torch==1.1? That is not correct. You should refer to pytorch.org.

That's not what's going on. The opening post details how pytorch was installed, and it's also in the logs linked above.

**Comment by peterjc123** [comment]
I extracted the build scripts from your log, that is, the code below.
```cmd
set PYTHON=Python36
set ARCH=-x64 
set PYTORCH=1.1.0-cp36-cp36m

git clone -q --depth=25 --branch=travis-pytorch-avy https://github.com/jvesely/PsyNeuLink.git C:\projects\psyneulink-wuxsn
cd C:\projects\psyneulink-wuxsn
git checkout -qf ae5a4dcbe1f72bf83c199ce8e95b21828b90219d

choco upgrade graphviz.portable -y
pip --version
pip install --user -U pip
pip --version
pip install --user -U certifi "numpy<1.16"
pip install --user git+https://github.com/benureau/leabra.git@master
if not "%PYTORCH%" == "" pip install --user http://download.pytorch.org/whl/cpu/torch-%PYTORCH%-win_amd64.whl
if "%PYTORCH%" == "" (findstr /V torch < dev_requirements.txt > tmp_req && move /Y tmp_req dev_requirements.txt)
pip install --user -e .[dev]
pytest --junit-xml=tests_out.xml -n auto --strict-markers tests/ %EXTRA_ARGS%
curl -X POST -F "file=@tests_out.xml" https://ci.appveyor.com/api/testresults/junit/%APPVEYOR_JOB_ID%
```
And then I tried to reproduce it locally, so I started to contrust the env according to your commands and did the smoke testing after that. And I found that it is throwing out an error.
```cmd
Python 3.6.2 (v3.6.2:5fd33b5, Jul  8 2017, 04:57:36) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python36\lib\site-packages\torch\__init__.py", line 79, in <module>
    from torch._C import *
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'
>>> quit()
```
After that, I tried to upgrade numpy from 3.15 to 3.16, and it worked. I also tried to downgrade torch to 1.0.1 and it also worked. So you could actually remove the version constraint on numpy and try again.

**Comment by jvesely** [comment]
Does it mean that pytorch-1.1.0 requires numpy>=3.16?
Shouldn't the wheel indicate its dependencies icnluding version?

There's a reason psyneulink restricts numpy to <3.16; change of behavior that causes errors in few workloads.
I've run appveyor tests and numpy-3.16 makes "import pytorch" succeed[0] (I can't test the full test run).

pytorch-1.1.0 appears to work OK with numpy-3.15 in Linux and macos [1].
Is windows wheel built in different python setting?

[0] https://ci.appveyor.com/project/jvesely/psyneulink-wuxsn/builds/25519000/job/frwx7pt4747a47nw?fullLog=true#L145
[1] https://travis-ci.org/PrincetonUniversity/PsyNeuLink/jobs/549300082

**Comment by peterjc123** [comment]
Yes, looks like the version constraints of numpy are inconsistent between wheels and conda packages and between different OS. @soumith Should we pin them to a specific version?
https://github.com/pytorch/builder/blob/master/wheel/build_wheel.sh#L133
https://github.com/pytorch/builder/blob/master/manywheel/build_common.sh#L99
https://github.com/pytorch/builder/blob/master/windows/build_pytorch.bat#L59
https://github.com/pytorch/builder/blob/master/conda/pytorch-nightly/meta.yaml#L15

**Comment by soumith** [comment]
@peterjc123 except for the windows one, the others all were pinned.

**Comment by jvesely** [comment]
@soumith will the fix be available for 1.1.0 wheels?
